### ISSUE TITLE
ViewWillAppear method getting called twice 

### ISSUE THREAD

**Issue by goelv** [romaonthego/RESideMenu#65]
Hi,

I am writing an application on iOS 7 using storyboards & the residemenu library. 

I have a navigation controller as my root view controller and I embed a view controller within it.

When the application loads, viewDidLoad gets called once and viewWillAppear gets called once. This is correct (also I check for data & update the view in view will appear).

However, in my view controller, i have a button which allows the user to share the information via SMS/text using the 'MFMessageComposeViewController' controller.

So the code here is

```
MFMessageComposeViewController *controller = [[MFMessageComposeViewController alloc] init];


if([MFMessageComposeViewController canSendText])

{

    controller.body = [NSString stringWithFormat:@"Vehicle parked at %@. Next cleaning date is in %@ days, %@",            address, daysRemaining, cleaningDateFormatted];

    controller.messageComposeDelegate = self;

    [self presentViewController:controller animated:YES completion:nil];
}
```

and the delegate function is

```
- (void)messageComposeViewController:(MFMessageComposeViewController *)controller didFinishWithResult:(MessageComposeResult)result
{
 switch (result) {
      case MessageComposeResultCancelled:
           NSLog(@"Cancelled");
           break;
      case MessageComposeResultSent:
           NSLog(@"Message Sent");
           break;
      default:
           break;
 }

[self dismissViewControllerAnimated:YES completion:nil];

}
```

However, when i called dismissViewControllerAnimated.. the viewDidAppear method is being called twice. Do you know why this is happening as this is checking data twice and there by adding things twice to my view instead of only once?

Thanks,
Varun
